hudi-examples/src/main/scala/org/apache/hudi/examples/spark/HoodieMorCompactionJob.scala

```diff
@@ -0,0 +1,110 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.examples.spark
+
+import org.apache.hudi.DataSourceWriteOptions.{PARTITIONPATH_FIELD_OPT_KEY, PRECOMBINE_FIELD_OPT_KEY, RECORDKEY_FIELD_OPT_KEY, TABLE_TYPE_OPT_KEY}
+import org.apache.hudi.QuickstartUtils.getQuickstartWriteConfigs
+import org.apache.hudi.client.SparkRDDWriteClient
+import org.apache.hudi.client.common.HoodieSparkEngineContext
+import org.apache.hudi.common.model.{HoodieAvroPayload, HoodieRecordPayload, HoodieTableType}
+import org.apache.hudi.common.util.Option
+import org.apache.hudi.config.HoodieWriteConfig.TABLE_NAME
+import org.apache.hudi.config.{HoodieCompactionConfig, HoodieWriteConfig}
+import org.apache.hudi.examples.common.{HoodieExampleDataGenerator, HoodieExampleSparkUtils}
+import org.apache.spark.sql.SaveMode.{Append, Overwrite}
+import org.apache.spark.sql.SparkSession
+
+import scala.collection.JavaConverters._
+
+/**
+ * Simple example to run a compaction job for MOR table.
+ * To run this example, you should:
+ *   1. For running in IDE, set VM options `-Dspark.master=local[2]`
+ *   2. For running in shell, using `spark-submit`
+ *
+ * Usage: HoodieMorCompactionJob <tablePath> <tableName>.
+ * <tablePath> and <tableName> describe root path of hudi and table name
+ * for example, `HoodieMorCompactionJob file:///tmp/hoodie/hudi_mor_table hudi_mor_table`
+ */
+object HoodieMorCompactionJob {
+
+  def main(args: Array[String]): Unit = {
+    if (args.length < 2) {
+      System.err.println("Usage: HoodieMorCompactionJob <tablePath> <tableName>")
+      System.exit(1)
+    }
+
+    val spark = HoodieExampleSparkUtils.defaultSparkSession("Hudi Spark compaction example")
+    val dataGen = new HoodieExampleDataGenerator[HoodieAvroPayload]
+    val tablePath = args(0)
+    val tableName = args(1)
+    insertData(spark, tablePath, tableName, dataGen, HoodieTableType.MERGE_ON_READ.name())
+    updateData(spark, tablePath, tableName, dataGen, HoodieTableType.MERGE_ON_READ.name())
+    val cfg = HoodieWriteConfig.newBuilder()
+      .withPath(tablePath)
+      .withSchema(HoodieExampleDataGenerator.TRIP_EXAMPLE_SCHEMA)
+      .withParallelism(2, 2)
+      .forTable(tableName)
+      .withCompactionConfig(HoodieCompactionConfig.newBuilder()
+        .withInlineCompaction(true)
+        .withMaxNumDeltaCommitsBeforeCompaction(1).build())
+      .build()
+    val client = new SparkRDDWriteClient[HoodieRecordPayload[Nothing]](new HoodieSparkEngineContext(spark.sparkContext), cfg)
+    val instant = client.scheduleCompaction(Option.empty())
+    val writeStatues = client.compact(instant.get())
+    client.commitCompaction(instant.get(), writeStatues, Option.empty())
+    client.clean()
+    client.close()
```

Comment:
use `try-with-resources` to make sure the client object must be closed finally.

Reply:
Added try-catch-finally. Looks like try-with-resources is not readily available in scala versions earlier than 2.13